### PR TITLE
E2-1815: download files directly from URL instead of API call

### PR DESCRIPTION
**Pull request by e7mac**
As per https://leanplum.atlassian.net/browse/E2-1815 (Proposal 1), we will be sending the files key with the URL for each file. If this exists, we want the SDK to use the URL directly to get these files.

### CODE REVIEW

File: AndroidSDKCore/src/main/java/com/leanplum/Leanplum.java [hunk 1]
```diff
@@ -2303,6 +2307,13 @@ public static void clearUserContent() {
     return featureFlagSet;
   }
 
+  @VisibleForTesting
+  public static Map<String, String> parseFileURLs(JSONObject response) {
+    JSONObject filesObject = response.optJSONObject(
```
Comment (hrishileanplum) [inline AndroidSDKCore/src/main/java/com/leanplum/Leanplum.java]: I think you need try catch here? Also null check, if the json key does not exists?

File: AndroidSDKCore/src/main/java/com/leanplum/internal/FileManager.java [hunk 2]
```diff
@@ -185,6 +189,10 @@ public static boolean fileExistsAtPath(String path) {
     return path != null && new File(path).exists();
   }
 
+  public static void setFilenameToURL(Map<String, String> filenameToURL) {
+    FileManager.filenameToURL = filenameToURL;
```
Comment (hrishileanplum) [inline AndroidSDKCore/src/main/java/com/leanplum/internal/FileManager.java]: Just similar to previous iOS note on naming. Lets name it filenamesToUrls

File: AndroidSDKCore/src/main/java/com/leanplum/internal/FileManager.java [hunk 3]
```diff
@@ -108,6 +109,9 @@ public static DownloadFileResult maybeDownloadFile(boolean isResource, String st
       if (!FileManager.fileExistsAtPath(realPath)) {
         realPath = FileManager.fileRelativeToDocuments(stringValue);
         if (!FileManager.fileExistsAtPath(realPath)) {
+          if (FileManager.filenameToURL != null && FileManager.filenameToURL.containsKey(stringValue) && urlValue == null) {
+            urlValue = FileManager.filenameToURL.get(stringValue);
```
Comment (e7mac) [inline AndroidSDKCore/src/main/java/com/leanplum/internal/FileManager.java]: I tried adding a test here, but the structure of code made it very difficult

File: AndroidSDKCore/src/main/java/com/leanplum/internal/FileManager.java (hunk 2 above)
Comment (hrishileanplum) [reply AndroidSDKCore/src/main/java/com/leanplum/internal/FileManager.java]: setFilenameToURL -? setFilenameToURLs

File: AndroidSDKCore/src/main/java/com/leanplum/Leanplum.java (hunk 1 above)
Comment (e7mac) [reply AndroidSDKCore/src/main/java/com/leanplum/Leanplum.java]: good "catch" :), thanks!

**Pull request merged**